versioning/BRM/dbrm.cpp

```diff
@@ -4453,6 +4454,97 @@ void DBRM::deleteAISequence(uint32_t OID)
   }
 }
 
+void DBRM::addToLBIDList(uint32_t sessionID, vector<LBID_t>& lbidList)
+{
+  boost::shared_ptr<execplan::CalpontSystemCatalog> systemCatalogPtr =
+    execplan::CalpontSystemCatalog::makeCalpontSystemCatalog(sessionID);
+
+  std::unordered_map<execplan::CalpontSystemCatalog::OID,
+    std::unordered_map<execplan::CalpontSystemCatalog::OID,
+      std::vector<struct BRM::EMEntry>>> extentMap;
+
+  int err = 0;
+
+  std::unordered_set<LBID_t> lbidSet;
+  std::unordered_set<execplan::CalpontSystemCatalog::OID> tableOidSet;
+
+  for (const auto i : lbidList)
+  {
+    lbidSet.insert(i);
+    execplan::CalpontSystemCatalog::OID oid;
+    uint16_t dbRoot, segmentNum;
+    uint32_t partitionNum, fbo;
+    err = lookupLocal(i, 0, false, oid, dbRoot, partitionNum, segmentNum, fbo);
+
+    if (err)
+    {
+      ostringstream os;
+      std::string errorMsg;
+      BRM::errString(err, errorMsg);
+      os << "lookupLocal from extent map error encountered while looking up lbid " << i
+         << " and error code is " << err << " with message " << errorMsg;
+      throw runtime_error(os.str());
+    }
+
+    execplan::CalpontSystemCatalog::OID tableOid =
+      systemCatalogPtr->isAUXColumnOID(oid);
+
+    if (tableOid >= 3000)
+    {
+      if (tableOidSet.find(tableOid) == tableOidSet.end())
+      {
+        tableOidSet.insert(tableOid);
+        execplan::CalpontSystemCatalog::TableName tableName =
+          systemCatalogPtr->tableName(tableOid);
+        execplan::CalpontSystemCatalog::RIDList tableColRidList =
+          systemCatalogPtr->columnRIDs(tableName);
+
+        for (unsigned j = 0; j < tableColRidList.size(); j++)
+        {
+          auto objNum = tableColRidList[j].objnum;
+          err = getExtents(objNum, extentMap[tableOid][objNum]);
+
+          if (err)
+          {
+            ostringstream os;
+            os << "BRM lookup error. Could not get extents for OID " << objNum;
+            throw runtime_error(os.str());
+          }
+        }
+      }
+
+      uint32_t extentNumAux = fbo / (execplan::AUX_COL_WIDTH * 1024);
```

Comment:
@tntnatbry can you please clarify a little bit on the code.
Looking at the comment, this function checks if the given lbids are "aux columnd" lbids, and then for table where particular aux column belongs it collects lbids for all other columns in that table.

1. You use "execplan::AUX_COL_WIDTH" the size of "normalized number of blocks", it's correct until 
getExtentRows() or BLOCK_SIZE are not changed, but I would like to suggest to make it more general, because if decide to change the "BLOCK_SIZE" or "Number of rows" in extent you code will give the wrong result.

```

2826   // Convert extent size in rows to extent size in 8192-byte blocks.
2827   // extentRows should be multiple of blocksize (8192).
2828   const unsigned EXTENT_SIZE = (getExtentRows() * colWidth) / BLOCK_SIZE;

//  size is the number of multiples of 1024 blocks.
2843   //  ex: size=1 --> 1024 blocks
2844   //      size=2 --> 2048 blocks
2845   //      size=3 --> 3072 blocks, etc.
2846   uint32_t size = EXTENT_SIZE / 1024;
```

2. As far as I know FBO specifies an offset in the segment file for the particular extent, in case we have a multiple extents per one segment file. That means each extent has a unique lbid, in this case I do not understand a check at the line 4527. Can you please clarify on it? 
Thanks.

Reply:
@denis0x0D 

For Point 1, this is a very good catch, thanks for letting me know. I have changed this line:
```c++
uint32_t extentNumAux = fbo / (execplan::AUX_COL_WIDTH * 1024);
```
to:
```c++
uint32_t extentNumAux = fbo / ((getExtentRows() * execplan::AUX_COL_WIDTH) / BLOCK_SIZE);
```

Regarding Point 2:

Here is the check:
```c++
          if (dbRoot == extent.dbRoot && partitionNum == extent.partitionNum &&                                                             
              segmentNum == extent.segmentNum && extentNumAux == extentNum)                                                                 
          { 
            lbidSet.insert(extent.range.start);                                                                                             
            break;                                                                                                                          
          }                                                                                                                                 
```

Are you saying we should not have the check `extentNumAux == extentNum`? If we remove this check, then in case the segment file contains multiple extents, we will end up inserting multiple starting lbids into `lbidSet` for a single aux column lbid, which will be incorrect. We only want the lbids for all other columns in the table which correspond to a given extent number in the segment file, not for all extents in that segment file. Does this clarify?

Reply:
Can you please add a comment how does actually mapping between aux column and user columns does work.

Reply:
@denis0x0D  I will mention this when I create a high-level design doc for this feature. I will let you know once I have it.